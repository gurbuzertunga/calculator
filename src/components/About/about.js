/* eslint-disable max-len */
const about = () => (
  <div className="hello">
    <h1 className=" mb-5">Welcome to our page</h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. IpsuLorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. IpsuWelcome to our page!HomeCalculatorQuote
    </p>
  </div>
);

export default about;
