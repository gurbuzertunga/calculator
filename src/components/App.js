/* eslint-disable */
import { useState } from 'react';
import Display from './display';
import ButtonPanel from './buttonPanel';
import calculate from '../logic/calculate';

function App() {
  let [total, setTotal] = useState('0');
  let [next, setNext] = useState(null);
  let [operation, setOperation] = useState(null);
  function handleClick(e) {
    if (e.target.classList.contains('math-btn')) {
      var result = calculate({ total , next, operation }, e.target.innerText);
      if (result.total || result.total === null ){
        setTotal(result.total)
        
      }
      if (result.operation || result.operation === null ){
        setOperation(result.operation)
      }
      if (result.next || result.next === null){
        setNext(result.next)
      }
      
    }
  }
  return (
    <div className="container d-flex flex-column app rounded">
      <Display next={next}  op= {operation} total={total} />
      <ButtonPanel handleClick={handleClick} />
    </div>
  );
}

export default App;
