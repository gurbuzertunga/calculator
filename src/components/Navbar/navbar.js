import { Link } from 'react-router-dom';

const navbar = () => (
  <div className="navbar w-100 nav bg-light p-4 ">
    <div className="left">
      <p className="mb-0 font-weight-bold h5 text-decoration-none">Math Magicians</p>
    </div>

    <div className="right ">
      <Link to="/" className="mr-5 h5 text-dark text-decoration-none home-btn ">Home</Link>
      <Link to="/calculator" className="mr-5 h5 text-dark text-decoration-none calculator-btn">Calculator</Link>
      <Link to="/quote" className="mr-5 h5 text-dark text-decoration-none quote-btn">Quote of the day</Link>
    </div>
  </div>
);
export default navbar;
