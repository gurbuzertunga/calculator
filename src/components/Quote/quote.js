const quote = () => (
  /* eslint-disable max-len */

  <div className="quote">
    <h1 className="text-center mb-5">Quote of the day</h1>
    <p className="h5">Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“William Paul Thurston</p>
  </div>
);

export default quote;
