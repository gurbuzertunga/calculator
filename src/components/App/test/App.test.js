import { render } from 'enzyme';
import { expect as chaiExpect } from 'chai';
import toJson from 'enzyme-to-json';
import App from '../App';
import Navbar from '../../Navbar/navbar';
import Display from '../../Display/display';
import About from '../../About/about';
import Quote from '../../Quote/quote';

describe('It works', () => {
  it('renders without crashing', () => {
    render(App);
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('contains navbar', () => {
    const wrapper = mount(<App />);
    chaiExpect(wrapper.contains(Navbar)).to.equal(true);
  });
  it('contains about', () => {
    const wrapper = mount(<App />);
    chaiExpect(wrapper.contains(About)).to.equal(true);
  });
  it('contains quote', () => {
    const wrapper = mount(<App />);
    chaiExpect(wrapper.contains(Quote)).to.equal(true);
  });
  it('contains display', () => {
    const wrapper = mount(<App />);
    chaiExpect(wrapper.contains(Display)).to.equal(true);
  });
});
