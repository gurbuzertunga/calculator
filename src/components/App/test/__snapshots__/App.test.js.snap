// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It works renders without crashing 1`] = `
<BrowserRouter>
  <navbar />
  <div
    className="container d-flex flex-column app rounded justify-content-center "
  >
    <Routes>
      <Route
        element={<About />}
        exact={true}
        path="/"
      />
      <Route
        element={
          <div
            className="calculator align-self-center"
          >
            <Display
              next={null}
              op={null}
              total="0"
            />
             
            <ButtonPanel
              handleClick={[Function]}
            />
             
          </div>
        }
        exact={true}
        path="/calculator"
      />
      <Route
        element={<quote />}
        exact={true}
        path="/quote"
      />
    </Routes>
  </div>
</BrowserRouter>
`;
